www/js/diary/list.js

```diff
@@ -487,16 +494,20 @@ angular.module('emission.main.diary.list',['ui-leaflet',
    {text:'Religious', value:'religious'},
    {text:'Other',value:'other_purpose'}];
 
-   $scope.storeMode = function(mode_val) {
+   $scope.storeMode = function(mode_val, isOther) {
       $scope.draftMode.label = mode_val;
       Logger.log("in storeMode, after setting mode_val = "+mode_val+", draftMode = "+JSON.stringify($scope.draftMode));
       $window.cordova.plugins.BEMUserCache.putMessage(MODE_CONFIRM_KEY, $scope.draftMode);
+      if(isOther = true) 
```

Comment:
isn't `isOther = true` assignment? Shouldn't it be `isOther == true` for equality?

Reply:
Sorry, I will make this change. This worked because it was just `$scope.draftPurpose = angular.undefined` after every store.